### PR TITLE
Add some documentation and debugging statements.

### PR DESCRIPTION
**Pull request by AndyZe**
minor stuff...

### CODE REVIEW

**Comment by AndyZe** [comment]
I would like it if you create a "kinetic" branch for the newest ROS release. Then I would do a PR for that branch, too.

**Comment by AndyZe** [comment]
Can I also ask, are the predictions very accurate for you? They are not accurate at all for me -- most of the predictions are "half speed." Maybe it's been trained for you, though.

File: nodes/voice_control_updated.py [hunk 1]
```diff
@@ -62,13 +66,22 @@ def start_recognizer(self):
...
-        stream = pyaudio.PyAudio().open(format=pyaudio.paInt16, channels=1,
-                        rate=16000, input=True, frames_per_buffer=1024)
+        rospy.loginfo("Opening the audio channel")
+
+	# I recommend installing and running audacity to help figure this out
+	# Other useful commands:
+	# pactl list short sources
+	# pacmd list-sinks
+        stream = pyaudio.PyAudio().open(format=pyaudio.paInt16, channels=2,
+                        rate=44100, input=True, frames_per_buffer=65536)
```
Comment (Pankaj-Baranwal) [inline nodes/voice_control_updated.py]: What effect did changing the values to these specific constants have?
Channels I can understand, but others?

**Review by Pankaj-Baranwal — APPROVED** [review]

**Comment by Pankaj-Baranwal** [comment]
Well, you are right. The predictions do not work correctly most of the times. 
In my case, if I speak at the right moments and with proper phonetics, the system is able to detect "right", "left", "half speed" much more accurately.
Currently, this repo uses the same recogniser as the original one.
I am currently working on improving the accuracy. But it might take some time before it is completed.

File: demo/voice_cmd.dic [hunk 2]
```diff
@@ -1,11 +1,9 @@
 BACK	B AE K
-BACKWARD	B AE K W ER D
 FORWARD	F AO R W ER D
 FULL	F UH L
 HALF	HH AE F
-HALT	HH AO L T
```
Comment (Pankaj-Baranwal) [inline demo/voice_cmd.dic:6]: Did you remove these two so as to avoid confusion between "BACK" <-> "BACKWARD" and "HALF" <-> "HALT"?

Comment (AndyZe) [reply demo/voice_cmd.dic:6]: Hmm, that wasn't an intentional change. I can revert it.

**Comment by AndyZe** [comment]
Thanks for reviewing. You've done the community a pretty good service by releasing a ROS Kinetic-compatible version, even if that wasn't intentional.  ;)

File: nodes/voice_control_updated.py (hunk 1 above)
Comment (AndyZe) [reply nodes/voice_control_updated.py]: Those are just the specific values for my computer's hardware. I reverted them back to your settings in the most recent.

**Comment by AndyZe** [comment]
I'm getting much better accuracy after changing the audio channel format. Read here:
http://cmusphinx.sourceforge.net/wiki/tutorialtuning

This is the change I made:
# Pocketsphinx requires 16kHz, mono, 16-bit little-Endian audio.
	# See http://cmusphinx.sourceforge.net/wiki/tutorialtuning
        stream = pyaudio.PyAudio().open(format=pyaudio.paInt16, channels=1,
rate=16000, input=True, frames_per_buffer=65536)

I'll make another pull request, if you want (I would prefer to have a Kinetic branch, though, because that's what I'm using)

**Comment by Pankaj-Baranwal** [comment]
Thanks for the feedback. And the article. I will read it soon.
I have my semester exams starting next week because of which I haven't been able to actively contribute here. I will try to update this repo with the Kinetic branch, possibly this weekend.

**Pull request merged**